frontend/src/modules/fluenteditor/components/Editor.js

```diff
@@ -146,7 +146,11 @@ export class EditorBase extends React.Component<EditorProps, State> {
             }
         }
 
-        props.setInitialTranslation(translationContent);
+        // Update the initial translation content only when the entity changed, not
+        // when new content is loaded from an external source.
+        if (typeof translation === 'undefined') {
+            props.setInitialTranslation(translationContent);
+        }
```

Comment:
The code change makes sense, but I'm unable to figure out what problem was the old code causing in the UI.

Reply:
Right, sorry, that's unrelated. I thought it was a cause of issue but it wasn't, and I didn't revert it. This was causing a failure in the unsaved changes code: if you load a translation from for example the History content, the initial translation would be updated to that translation, and thus when leaving the entity, the unsaved changes warning would not be triggered. Do you want me to move this to a different PR?

Reply:
Nope, feel free to leave this in. I'm unable to reproduce the problem, though:
https://pontoon.mozilla.org/sl/firefox/all-resources/?string=203763